deployment/hasura/migrations/Aerie/22_plan_model_migration/up.sql

```diff
@@ -0,0 +1,526 @@
+		on conflict (directive_id, plan_id, tag_id) do nothing;
+
+		-- Presets
+		delete from merlin.preset_to_directive
+		  where plan_id = _plan_id;
+		insert into merlin.preset_to_directive(preset_id, activity_id, plan_id)
+			select pts.preset_id, pts.activity_id, _plan_id
+			from merlin.preset_to_snapshot_directive pts
+			where pts.snapshot_id = _snapshot_id
+			on conflict (activity_id, plan_id)
+			do update	set preset_id = excluded.preset_id;
+
+		-- Clean up
+		drop table diff;
+  end
+$$;
+
+comment on procedure merlin.restore_from_snapshot(_plan_id integer, _snapshot_id integer) is e''
+	'Restore a plan to its state described in the given snapshot.';
+
+-- Update create_merge_request function to check model IDs
+create or replace function merlin.create_merge_request(plan_id_supplying integer, plan_id_receiving integer, request_username text)
+  returns integer
+  language plpgsql as $$
+declare
+  merge_base_snapshot_id integer;
+  validate_planIds integer;
+  supplying_snapshot_id integer;
+  merge_request_id integer;
+  model_id_receiving integer;
+  model_id_supplying integer;
+begin
+  if plan_id_receiving = plan_id_supplying then
+    raise exception 'Cannot create a merge request between a plan and itself.';
+  end if;
+  select id from merlin.plan where plan.id = plan_id_receiving into validate_planIds;
+  if validate_planIds is null then
+    raise exception 'Plan receiving changes (Plan %) does not exist.', plan_id_receiving;
+  end if;
+  select id from merlin.plan where plan.id = plan_id_supplying into validate_planIds;
+  if validate_planIds is null then
+    raise exception 'Plan supplying changes (Plan %) does not exist.', plan_id_supplying;
+  end if;
+
+  select merlin.create_snapshot(plan_id_supplying) into supplying_snapshot_id;
+
+  select merlin.get_merge_base(plan_id_receiving, supplying_snapshot_id) into merge_base_snapshot_id;
+  if merge_base_snapshot_id is null then
+    raise exception 'Cannot create merge request between unrelated plans.';
+  end if;
+
+  select model_id from merlin.plan where plan.id = plan_id_receiving into model_id_receiving;
+  select model_id from merlin.plan where plan.id = plan_id_supplying into model_id_supplying;
+  if model_id_receiving is distinct from model_id_supplying then
+    raise exception 'Cannot create merge request: plan supplying changes is using a different model (%) than the receiving plan (%)', model_id_supplying, model_id_receiving;
+  end if;
+
+  insert into merlin.merge_request(plan_id_receiving_changes, snapshot_id_supplying_changes, merge_base_snapshot_id, requester_username)
+    values(plan_id_receiving, supplying_snapshot_id, merge_base_snapshot_id, request_username)
+    returning id into merge_request_id;
+  return merge_request_id;
+end
+$$;
+
+-- Alter set_revisions_and_initialize_dataset_on_insert
+-- drop trigger first & recreate after
+drop trigger set_revisions_and_initialize_dataset_on_insert_trigger on merlin.simulation_dataset;
```

Comment:
This `drop trigger` and the `create trigger` on L276 are not needed. Similarly, in the `down.sql`, the statements on L314 and L346 aren't needed.